okhttp/src/main/java/okhttp3/internal/http2/Http2Stream.kt

```diff
@@ -13,321 +13,291 @@
+      throw errorException ?: StreamResetException(errorCode!!)
     }
-    if (!source.finished || !source.receiveBuffer.exhausted() || !source.readBuffer.exhausted()) {
-      throw new IllegalStateException("too early; can't read the trailers yet");
+    check(source.finished && source.receiveBuffer.exhausted() && source.readBuffer.exhausted()) {
+      "too early; can't read the trailers yet"
     }
-    return source.trailers != null ? source.trailers : Util.EMPTY_HEADERS;
-  }
-
-  /**
-   * Returns the reason why this stream was closed, or null if it closed normally or has not yet
-   * been closed.
-   */
-  public synchronized ErrorCode getErrorCode() {
-    return errorCode;
+    return source.trailers ?: Util.EMPTY_HEADERS
   }
 
   /**
    * Sends a reply to an incoming stream.
    *
    * @param outFinished true to eagerly finish the output stream to send data to the remote peer.
-   *     Corresponds to {@code FLAG_FIN}.
+   *     Corresponds to `FLAG_FIN`.
    * @param flushHeaders true to force flush the response headers. This should be true unless the
    *     response body exists and will be written immediately.
    */
-  public void writeHeaders(List<Header> responseHeaders, boolean outFinished, boolean flushHeaders)
-      throws IOException {
-    assert (!Thread.holdsLock(Http2Stream.this));
-    if (responseHeaders == null) {
-      throw new NullPointerException("headers == null");
-    }
-    synchronized (this) {
-      this.hasResponseHeaders = true;
+  @Throws(IOException::class)
+  fun writeHeaders(responseHeaders: List<Header>, outFinished: Boolean, flushHeaders: Boolean) {
+    var flushHeaders = flushHeaders
+    assert(!Thread.holdsLock(this@Http2Stream))
+    synchronized(this) {
+      this.hasResponseHeaders = true
       if (outFinished) {
-        this.sink.finished = true;
+        this.sink.finished = true
       }
     }
 
     // Only DATA frames are subject to flow-control. Transmit the HEADER frame if the connection
     // flow-control window is fully depleted.
     if (!flushHeaders) {
-      synchronized (connection) {
-        flushHeaders = connection.bytesLeftInWriteWindow == 0L;
+      synchronized(connection) {
+        flushHeaders = connection.bytesLeftInWriteWindow == 0L
       }
     }
 
-    connection.writeHeaders(id, outFinished, responseHeaders);
+    connection.writeHeaders(id, outFinished, responseHeaders)
 
     if (flushHeaders) {
-      connection.flush();
+      connection.flush()
     }
   }
 
-  public void enqueueTrailers(Headers trailers) {
-    synchronized (this) {
-      if (sink.finished) throw new IllegalStateException("already finished");
-      if (trailers.size() == 0) throw new IllegalArgumentException("trailers.size() == 0");
-      this.sink.trailers = trailers;
+  fun enqueueTrailers(trailers: Headers) {
+    synchronized(this) {
+      check(!sink.finished) { "already finished" }
+      check(trailers.size() != 0) { "trailers.size() == 0" }
```

Comment:
nit: `require` for `IllegalArgumentException`

Reply:
Good catch.